src/interface_elements.jl

```diff
@@ -0,0 +1,546 @@
+    basedofs = get_dofs(getproperty(ip, side))
+    if isempty(basedofs)
+        return (Tuple{}(),)
+    else
+        return broadcast.(get_interface_index, ((ip,),), ((side,),), basedofs)
+    end
+end
+
+vertexdof_indices(ip::InterfaceCellInterpolation) = get_interface_dof_indices(vertexdof_indices, ip)
+
+function facedof_indices(ip::InterfaceCellInterpolation)
+    here = (get_interface_dof_indices(vertexdof_indices, ip, :here)...,
+            get_interface_dof_indices(facedof_interior_indices, ip, :here)...,
+            get_interface_dof_indices(celldof_interior_indices, ip, :here)...)
+    there = (get_interface_dof_indices(vertexdof_indices, ip, :there)...,
+             get_interface_dof_indices(facedof_interior_indices, ip, :there)...,
+             get_interface_dof_indices(celldof_interior_indices, ip, :there)...)
+    return Tuple(vcat(collect( [t...] for t in here )...)), Tuple(vcat(collect( [t...] for t in there )...))
+end
+
+facedof_interior_indices(ip::InterfaceCellInterpolation) = get_interface_dof_indices(celldof_interior_indices, ip)
+
+edgedof_indices(ip::InterfaceCellInterpolation) = get_interface_dof_indices(facedof_indices, ip)
+
+edgedof_interior_indices(ip::InterfaceCellInterpolation) = get_interface_dof_indices(facedof_interior_indices, ip)
+
+function adjust_dofs_during_distribution(ip::InterfaceCellInterpolation)
+    return adjust_dofs_during_distribution(ip.here) || adjust_dofs_during_distribution(ip.there) # TODO: Is this really the way to do it?
+end
+
+n_components(ip::InterfaceCellInterpolation) = n_components(ip.here)
+
+function default_interpolation(::Type{InterfaceCell{sdim, rdim, shape, rshape, Chere, Cthere}}
+    ) where {sdim, rdim, shape<:AbstractRefShape{sdim}, rshape<:AbstractRefShape{rdim},
+             Chere<:AbstractCell{rshape}, Cthere<:AbstractCell{rshape}} 
+    return InterfaceCellInterpolation(default_interpolation(Chere), default_interpolation(Cthere))
+end
+
+function default_geometric_interpolation(ip::InterfaceCellInterpolation{sdim, <:Any, <:Any, <:Any, IPhere, IPthere}
+    ) where {sdim, IPhere<:ScalarInterpolation, IPthere<:ScalarInterpolation}
+    return InterfaceCellInterpolation(ip.here, ip.there)^sdim
+end
+function default_geometric_interpolation(ip::InterfaceCellInterpolation{sdim, <:Any, <:Any, <:Any, IPhere, IPthere}
+    ) where {sdim, IPhere<:VectorizedInterpolation, IPthere<:VectorizedInterpolation}
+    return InterfaceCellInterpolation(ip.here.ip, ip.there.ip)^sdim
+end
+function default_geometric_interpolation(ip::VectorizedInterpolation{<:Any, <:Any, <:Any, <:InterfaceCellInterpolation})
+    return ip
+end
+
+getorder(ip::InterfaceCellInterpolation) = getorder(ip.here) == getorder(ip.there) ? getorder(ip.here) : (getorder(ip.here), getorder(ip.there))
+getorder(ip::VectorizedInterpolation{<:Any,<:Any,<:Any,<:InterfaceCellInterpolation}) = getorder(ip.ip)
+
+Base.:(^)(ip::InterfaceCellInterpolation, vdim::Int) = VectorizedInterpolation{vdim}(ip)
```

Comment:
I don't think this definition is needed since `InterfaceCellInterpolation <: ScalarInterpolation`